crates/apollo-compiler/src/ast/serialize.rs

```diff
@@ -829,6 +845,89 @@ fn serialize_string_value(state: &mut State, mut str: &str) -> fmt::Result {
     state.write("\"")
 }
 
+fn serialize_block_string(state: &mut State, str: &str) -> fmt::Result {
+    const TRIPLE_QUOTE: &str = "\"\"\"";
+    const ESCAPED_TRIPLE_QUOTE: &str = "\\\"\"\"";
+    const _: () = assert!(TRIPLE_QUOTE.len() == 3);
+    const _: () = assert!(ESCAPED_TRIPLE_QUOTE.len() == 4);
+
+    state.write(TRIPLE_QUOTE)?;
+    // `can_be_block_string` excludes \r, so the only remaining line terminator is \n
+    for mut line in str.split('\n') {
+        if line.is_empty() {
+            // Skip indentation which would be trailing whitespace
+            state.write("\n")?;
+            continue;
+        }
+        state.require_new_line()?;
+        loop {
+            if let Some((before, after)) = line.split_once(TRIPLE_QUOTE) {
+                state.write(before)?;
+                state.write(ESCAPED_TRIPLE_QUOTE)?;
+                line = after;
+            } else {
+                state.write(line)?;
+                break;
+            }
+        }
+    }
+    state.require_new_line()?;
+    state.write(TRIPLE_QUOTE)
+}
+
+fn prefer_block_string(value: &str) -> bool {
+    // If a value is multi-line, it likely looks nicer as a block string
```

Comment:
Maybe also if the value has `"` in it?
```graphql
"""Surrounded by "double" quotes"""
```

Reply:
my 5 euros (inflation) is that we should do what graphql js does, otherwise each implementation starts having their own SDL display 😅